wp-parsely.php

```diff
@@ -73,34 +73,10 @@
 const PARSELY_VERSION = '3.16.0';
 const PARSELY_FILE    = __FILE__;
 
-require_once __DIR__ . '/src/Models/class-base-model.php';
-require_once __DIR__ . '/src/Models/class-smart-link.php';
-require_once __DIR__ . '/src/Models/class-inbound-smart-link.php';
-
-require_once __DIR__ . '/src/class-parsely.php';
-require_once __DIR__ . '/src/class-permissions.php';
-require_once __DIR__ . '/src/class-scripts.php';
-require_once __DIR__ . '/src/class-dashboard-link.php';
-require_once __DIR__ . '/src/class-validator.php';
-require_once __DIR__ . '/src/UI/class-admin-bar.php';
-require_once __DIR__ . '/src/UI/class-metadata-renderer.php';
-require_once __DIR__ . '/src/Endpoints/class-metadata-endpoint.php';
-require_once __DIR__ . '/src/Endpoints/class-graphql-metadata.php';
-require_once __DIR__ . '/src/Telemetry/telemetry-init.php';
-
-require_once __DIR__ . '/src/class-metadata.php';
-require_once __DIR__ . '/src/Metadata/class-metadata-builder.php';
-require_once __DIR__ . '/src/Metadata/class-author-archive-builder.php';
-require_once __DIR__ . '/src/Metadata/class-category-builder.php';
-require_once __DIR__ . '/src/Metadata/class-date-builder.php';
-require_once __DIR__ . '/src/Metadata/class-front-page-builder.php';
-require_once __DIR__ . '/src/Metadata/class-page-builder.php';
-require_once __DIR__ . '/src/Metadata/class-page-for-posts-builder.php';
-require_once __DIR__ . '/src/Metadata/class-paginated-front-page-builder.php';
-require_once __DIR__ . '/src/Metadata/class-post-builder.php';
-require_once __DIR__ . '/src/Metadata/class-tag-builder.php';
+if ( file_exists( __DIR__ . '/vendor/autoload.php' ) ) {
+	require __DIR__ . '/vendor/autoload.php';
+}
 
-add_action( 'plugins_loaded', __NAMESPACE__ . '\\parsely_initialize_plugin' );
```

Comment:
Moving these to below the function makes it harder to read. You need to know the context that the function is being called with, to have the parameters make sense.

Reply:
Thanks for the feedback! That's a fair point that I didn't consider. 

However, I feel that with the DocBlocks in between the `add_action` and the function definition, it breaks the flow and - personally speaking - makes it a bit more difficult to follow. I feel that having the `add_action`/`add_filter` right below the callback function makes it a bit easier to read, even though we can lose the context of the function parameters, specially on larger functions. 

What do you think @GaryJones? I'm happy to revert back to the original organization, if we think it's an arguably better developer experience.

Reply:
I've been advocating for filter, documentation, function, since at least 2014 (Tom wrote https://tommcfarlin.com/clean-wordpress-code/ after we discussed it). The docblock can be mentally skipped when reading the code, so having the parameter context, then what is happening to it, makes more sense than having a function that does something to some values, and only then finding out what the values are afterwards.

Reply:
@GaryJones thanks for sharing that blog post, super interesting! 

> The docblock can be mentally skipped when reading the code

I do agree, although personally I do find slightly more difficult to read through a large file and parse what is happening when I have the docblocks in the middle. But I admit that might be the way my brain works. With that perspective in mind, I think it will make it easier to read as I can logically understand why the filter is above the documentation, and not below the callback, so thanks for sharing it!

Although, playing devil's advocate here, one could argue that with the documentation - function - filter format, all the required information about the function parameters can be found on the docblock, and/or inferred from the parameter’s name.

Anyway, in this case, let's not fix what isn't broken 😄  I will revert that change and go back to the former filter - documentation - function format!

Again, thank you for that context, was a very interesting read and something I will keep on my mind.